homeassistant/components/zha/sensor.py

```diff
@@ -496,35 +504,81 @@ class SmartEnergyMetering(PollableSensor):
 
     _use_custom_polling: bool = False
     _attribute_name = "instantaneous_demand"
-    _attr_device_class: SensorDeviceClass = SensorDeviceClass.POWER
-    _attr_state_class: SensorStateClass = SensorStateClass.MEASUREMENT
     _attr_translation_key: str = "instantaneous_demand"
 
-    unit_of_measure_map = {
-        0x00: UnitOfPower.WATT,
-        0x01: UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR,
-        0x02: UnitOfVolumeFlowRate.CUBIC_FEET_PER_MINUTE,
-        0x03: f"100 {UnitOfVolumeFlowRate.CUBIC_METERS_PER_HOUR}",
-        0x04: f"US {UnitOfVolume.GALLONS}/{UnitOfTime.HOURS}",
-        0x05: f"IMP {UnitOfVolume.GALLONS}/{UnitOfTime.HOURS}",
-        0x06: UnitOfPower.BTU_PER_HOUR,
-        0x07: f"l/{UnitOfTime.HOURS}",
-        0x08: UnitOfPressure.KPA,  # gauge
-        0x09: UnitOfPressure.KPA,  # absolute
-        0x0A: f"1000 {UnitOfVolume.GALLONS}/{UnitOfTime.HOURS}",
-        0x0B: "unitless",
-        0x0C: f"MJ/{UnitOfTime.SECONDS}",
+    _map = {
+        0x00: SmartEnergyMeteringClassMap(
+            UnitOfPower.WATT, SensorDeviceClass.POWER, SensorStateClass.MEASUREMENT
+        ),
+        0x01: SmartEnergyMeteringClassMap(
+            UnitOfVolume.CUBIC_METERS,
+            SensorDeviceClass.VOLUME,
+            SensorStateClass.TOTAL_INCREASING,
+        ),
+        0x02: SmartEnergyMeteringClassMap(
+            UnitOfVolumeFlowRate.CUBIC_FEET_PER_MINUTE,
+            SensorDeviceClass.VOLUME,
+            SensorStateClass.TOTAL_INCREASING,
+        ),
+        0x03: SmartEnergyMeteringClassMap(
+            f"100 {UnitOfVolume.CUBIC_FEET}",
```

Comment:
You should really add some scaling field here. This is a non standard unit  for the device class. I think that will even error. Yes i do see it is like this already, but still..

Reply:
Will do 👍

Reply:
Added a scaling field. Fixed also some state & device class I copied wrong.